model/map/src/main/java/org/keycloak/models/map/authorization/MapResourceServerStore.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.authorization;
+
+import org.jboss.logging.Logger;
+import org.keycloak.authorization.AuthorizationProvider;
+import org.keycloak.authorization.model.PermissionTicket;
+import org.keycloak.authorization.model.Policy;
+import org.keycloak.authorization.model.Resource;
+import org.keycloak.authorization.model.ResourceServer;
+import org.keycloak.authorization.model.Scope;
+import org.keycloak.authorization.store.PermissionTicketStore;
+import org.keycloak.authorization.store.PolicyStore;
+import org.keycloak.authorization.store.ResourceServerStore;
+import org.keycloak.authorization.store.ResourceStore;
+import org.keycloak.authorization.store.ScopeStore;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ModelDuplicateException;
+import org.keycloak.models.ModelException;
+import org.keycloak.models.map.authorization.adapter.MapResourceServerAdapter;
+import org.keycloak.models.map.authorization.entity.MapResourceServerEntity;
+import org.keycloak.models.map.common.Serialization;
+import org.keycloak.models.map.storage.MapKeycloakTransaction;
+import org.keycloak.models.map.storage.MapStorage;
+import org.keycloak.storage.StorageId;
+
+import static org.keycloak.common.util.StackUtil.getShortStackTrace;
+
+public class MapResourceServerStore implements ResourceServerStore {
+
+    private static final Logger LOG = Logger.getLogger(MapResourceServerStore.class);
+    private final KeycloakSession session;
+    private final AuthorizationProvider authorizationProvider;
+    final MapKeycloakTransaction<String, MapResourceServerEntity, ResourceServer> tx;
```

Comment:
I am not sure we can change this to String easily. Currently, we create a ResourceServer entity for each client that has authorization enabled and we are using clientId as id for the resourceServer. So we cannot say anything about the structure of clientId. Maybe we can do some migration step to change id to some random UUID and store clientId as some property, but I think it may require a lot of changes so maybe we should do it in a separate PR.